src/test/py/bazel/bazel_windows_test.py

```diff
@@ -397,6 +397,48 @@ def testRunWithScriptPath(self):
     self.AssertExitCode(exit_code, 0, stderr)
     self.assertIn('Hello from test!', '\n'.join(stdout))
 
+  def testZipUndeclaredTestOutputs(self):
+    self.CreateWorkspaceWithDefaultRepos('WORKSPACE')
+    self.ScratchFile('BUILD', [
+        'sh_test(',
+        '  name = "foo_test",',
+        '  srcs = ["foo.sh"],',
+        ')',
+        '',
+    ])
+    self.ScratchFile('foo.sh', [
+        'touch "$TEST_UNDECLARED_OUTPUTS_DIR/foo.txt"',
+    ])
+
+    exit_code, stdout, stderr = self.RunBazel(['info', 'bazel-testlogs'])
+    self.AssertExitCode(exit_code, 0, stderr)
+    bazel_testlogs = stdout[0]
+
+    output_file = os.path.join(bazel_testlogs, 'foo_test/test.outputs/foo.txt')
+    output_zip = os.path.join(bazel_testlogs,
+                              'foo_test/test.outputs/outputs.zip')
+
+    # Run the test with undeclared outputs zipping.
+    exit_code, _, stderr = self.RunBazel([
+        'test',
+        '--zip_undeclared_test_outputs',
+        '//:foo_test',
+    ], )
+    self.AssertExitCode(exit_code, 0, stderr)
+    # FIXME: The Windows test runner does not delete the undeclared outputs
+    #  after zipping, which differs from the behavior on other platforms.
```

Comment:
Fixing this would require implementing recursive directory removal in the Windows runner. The code already exists in `src/main/cpp/util/file_windows.cc`, let me know if you want me to copy it over.